pandas/core/internals/blocks.py

```diff
@@ -663,7 +663,9 @@ def replace(
         regex = should_use_regex(regex, to_replace)
 
         if regex:
-            return self._replace_regex(to_replace, value, inplace=inplace)
+            self.values = np.asarray(
```

Comment:
I do not think that this is the right place. Please try doing this deeper when operating with actual arrays